### DISCUSSION TITLE
A few questions about the DiehlAndCook2015 model

### DISCUSSION THREAD

**Discussion by rafaelblevin821** [BindsNET/bindsnet#646] [General]
Hi, I just have a few questions about the DiehlAndCook2015 model that I am trying to understand.

```
# Layers
        input_layer = Input(
            n=self.n_inpt, shape=self.inpt_shape, traces=True, tc_trace=20.0
        )
        exc_layer = DiehlAndCookNodes(
            n=self.n_neurons,
            traces=True,
            rest=-65.0,
            reset=-60.0,
            thresh=-52.0,
            refrac=5,
            tc_decay=100.0,
            tc_trace=20.0,
            theta_plus=theta_plus,
            tc_theta_decay=tc_theta_decay,
        )
        inh_layer = LIFNodes(
            n=self.n_neurons,
            traces=False,
            rest=-60.0,
            reset=-45.0,
            thresh=-40.0,
            tc_decay=10.0,
            refrac=2,
            tc_trace=20.0,
        )
```
I know the documentation briefly describes the parameters, but I'm still unsure how they are implemented.

I am particularly interested in the parameters `reset`, `tc_decay`, and `tc_trace`. 
Why is a `reset` value used when we have a resting potential value? 
If `tc_decay` is 100 by default, does that not limit the number of spikes to <4 if the simulation time is, say, 500? 
Why is `tc_trace` needed when the tc_decay is in place? 

Also, are these all callable arguments from the script that can be changed if necessary?

I'd be grateful for any insight on this.

**Comment by Hananel-Hazan** [comment]
> I know the documentation briefly describes the parameters, but I'm still unsure how they are implemented.

The implementation of the DiehlAndCookNodes neuron can be found [here](https://github.com/BindsNET/bindsnet/blob/9a2cc55fb223cd05efdaf35eacee655a2c8c9c5d/bindsnet/network/nodes.py#L981) and the implementation of the LIFNodes can be found [here](https://github.com/BindsNET/bindsnet/blob/9a2cc55fb223cd05efdaf35eacee655a2c8c9c5d/bindsnet/network/nodes.py#L418), you can find an answers to most of your questions. 

In brief, the code give you an option to use two different resting Vmem one for Post-spike reset and the other for rest value that use to calculate the decaying of the Vmem during inactivity. You can use the two values if you need or set them to the same value if you dont need them.

**Reply by rafaelblevin821** [reply]
Hi @Hananel-Hazan thanks. That clears up the `rest` and `reset` differences. 

Can you tell me the difference between `tc_decay` and `tc_trace`?

There is a one-line description in the documentation of each, but it is not very clear. Thanks.

**Reply by rafaelblevin821** [reply]
@Hananel-Hazan, sorry, scratch that. I think I've understood the differences now.

Please correct me if I am wrong, but tc_decay is the decay time of the spike to reset or rest, while tc_trace is the time that spike trace history remains to be used for calculations in weight updates. Is this correct?

**Reply by Hananel-Hazan** [reply]
Yes, that is correct
